TechnicalDocuments/FastlaneMatch.md

```diff
@@ -0,0 +1,147 @@
+
+## Update Contents Manually
+
+Apple does not allow more than two distribution certificates per account, so it can happen that fastlane match cannot create a new distribution certificate and is unable to figure out which existing certificate to link with a new provisioning profile. Fastlane match provides a utility for nuking all existing certificates and provisioning profiles and replace them with new ones generated by fastlane match. This is, however, not always advisable. Nuking existing enterprise certificates can be a really bad idea as in house apps will stop working.
+
+Since the files in the GitHub repository are encrypted they cannot be updated directly. Fastlane Match provides a couple of helper functions. To clone the repository, open a ruby console and call `Match::GitHelper.clone`. Babylon specific credentials can be found in the iOS 1Password vault.
+
+```shell
+$ bundle console
+irb(main):001:0> require 'match'
+=> true
+irb(main):002:0> branch = 'build_target'
+=> "build_target"
+irb(main):003:0> git_url = 'https://github.com/path/to/fastlane/match/repo'
+=> "https://github.com/path/to/fastlane/match/repo"
+irb(main):004:0> shallow_clone = false
+=> false
+irb(main):005:0> password = 'long-and-super-secret'
+=> "long-and-super-secret"
+irb(main):006:0> workspace = Match::GitHelper.clone(git_url, shallow_clone, manual_password: password, branch: branch)
+[14:22:10]: Cloning remote git repo…
+[14:22:10]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
+[14:22:12]: Checking out branch build_target…
+[14:22:12]: 🔓  Successfully decrypted certificates repo
+=> "/var/folders/8k/r0x0ys_927q8vq5_tq01ntjd2b03m3/T/d20190228-92193-16w4fz2"
+```
+
+Open another terminal session and navigate to the folder name returned by `Match::GitHelper.clone`. Once all edits are complete, return to the ruby console.
+
+```shell
+irb(main):007:0> commit_message = 'Updated provisioning profiles to be great again.'
+=> "Updated provisioning profiles to be great again.”
+irb(main):008:> Match::GitHelper.commit_changes(workspace, commit_message, git_url, branch)
+[14:28:44]: 🔒  Successfully encrypted certificates repo
+[14:28:44]: Pushing changes to remote git repo…
+=> nil
+```
+
+It can be convenient to store the password in the environment hash
+
+```shell
+irb(main):008:> ENV[“MATCH_PASSWORD"] = 'long-and-super-secret'
+```
+
+The password itself can be found in the iOS team 1Password vault.
+
+Certificates are stored in three folders
+
+```shell
+ certs/development
+ certs/distribution
+ certs/enterprise
+```
+
+It might be necessary to manually make a certificate available in both the enterprise and distribution folder. Copying the files did not work (February 2019), but adding a soft link did.
```

Comment:
This is very tricky advice. Enterprise certificates are only available for enterprise teams, distribution (AppStore) - only for "regular" teams. Why and how would one use the same certificate for enterprise and AppStore builds then?

Reply:
Ad hoc builds.